### ISSUE TITLE
Add sanity checking to constructor numsamples

### ISSUE THREAD

**Issue by tablatronix** [provideyourown/statistics#1]
Can have crashing if bad data makes into into here, might be useful to have some sanity checking here

**Comment by tablatronix** [comment]
Also get crashes in mean(), might need some sanity checks all around, ill PR if I get time

**Comment by tablatronix** [comment]
Looks like its stdDeviation that blows up because variance can be NAN.

as mRefVariance approaches sqr(mean()) variance can be negative

```
add data 23.75
variance()
...
return (5640.81 - (10 * 564.08)) / 10
return (5640.79 - (10 * 564.08)) / 10 // -0.001
return nan
```

**Comment by CaptainMack** [comment]
@tablatronix how do you suggest this can be fixed? I am having similar problems with std.Dev returning nan (I believe actually also mean, so might be a separate problem)

**Comment by tablatronix** [comment]
Sorry I totally forgot about this, I think I just patched it or made a wrapper that catches it , Ill try to look again and see.

**Comment by CaptainMack** [comment]
Hi @tablatronix just chiming in to see if you found the patch! I just had another case of std returning NaN ;-)

**Comment by tablatronix** [comment]
Yeah I did not fix it, I just check for isnan() in my code, I will see if I can undo that and reproduce again sinc eits been so long.
